api/src/test/java/org/openmrs/module/fhir2/BaseFhirContextSensitiveTest.java

```diff
@@ -9,19 +9,16 @@
  */
 package org.openmrs.module.fhir2;
 
-import org.mockito.Mockito;
-import org.openmrs.module.fhir2.api.util.LocalDateTimeFactory;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Primary;
+import org.junit.Before;
+import org.openmrs.module.fhir2.api.util.FhirGlobalPropertyHolder;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.test.context.ContextConfiguration;
 
-@Configuration
-public class MockedCalendarFactoryConfiguration {
+@ContextConfiguration(classes = { TestFhirSpringConfiguration.class }, inheritLocations = false)
+public abstract class BaseFhirContextSensitiveTest extends BaseModuleContextSensitiveTest {
 	
-	@Bean
-	@Primary
-	public LocalDateTimeFactory getCalendarFactory() {
-		return Mockito.mock(LocalDateTimeFactory.class);
+	@Before
+	public void setupBaseFhirContextSensitive() {
+		FhirGlobalPropertyHolder.reset();
 	}
```

Comment:
This came through odd, but I removed the MockedCalendarFactoryConfiguration class and took a different approach in the tests that required it.

And then I set up a base class for all of the context-sensitive tests in order to ensure the FhirGlobalPropertyHolder is reset before each test.

I believe this FhirGlobalPropertyHolder, in the end, was likely the primary cause of all of the issues, but I still think it's worth cleaning out any mocks from context sensitive tests, and trying to avoid static mocks if at all possible.